libs/wingsdk/src/target-tf-aws/api.ts

```diff
@@ -390,70 +390,105 @@ class WingRestApi extends Construct {
     super(scope, id);
     this.region = (App.of(this) as App).region;
 
-    const defaultResponse = API_CORS_DEFAULT_RESPONSE(props.cors);
+    // const defaultResponse = API_CORS_DEFAULT_RESPONSE(props.cors);
+
+    // let CORS = true;
+
+    let OAS30: OpenApiSpec = {
+      openapi: "3.0.1",
+      info: {
+        title: "fugazi",
+        version: "v1",
+      },
+      paths: {
+        "/$default": {
+          "x-amazon-apigateway-any-method": {
+            isDefaultRoute: true,
+            "x-amazon-apigateway-integration": {
+              payloadFormatVersion: "1.0",
+              type: "http_proxy",
+              httpMethod: "ANY",
+              uri: "https://example.com/",
+              connectionType: "INTERNET",
+            },
+          },
+        },
+      },
+      "x-amazon-apigateway-importexport-version": "1.0",
+    };
+
+    // let OAS30 = CORS
+    //   ? {
+    //       ...OAS30_BASE,
+    //       "x-amazon-apigateway-cors": {
+    //         allowMethods: ["GET", "OPTIONS", "POST"],
+    //         allowHeaders: ["*"],
+    //         maxAge: 0,
+    //         allowCredentials: false,
+    //         allowOrigins: ["*"],
+    //       },
+    //     }
+    //   : OAS30_BASE;
+
+    const apigwCorsConfig = props.cors
+      ? {
+          corsConfiguration: {
+            // renamed allowOrigin to allowOrigins
+            allowOrigins: props.cors.allowOrigins
+              ? props.cors.allowOrigins
+              : ["*"],
+            allowMethods: props.cors.allowMethods
+              ? props.cors.allowMethods
+              : ["GET", "POST", "PUT", "DELETE", "HEAD", "OPTIONS"],
+            allowHeaders: props.cors.allowHeaders
+              ? props.cors.allowHeaders
+              : ["Content-Type", "Authorization", "X-Requested-With"],
+            exposeHeaders: props.cors.exposeHeaders
+              ? props.cors.exposeHeaders
+              : [],
+            allowCredentials: props.cors.allowCredentials
+              ? props.cors.allowCredentials
+              : false,
+            // added maxAge
+            maxAge: props.cors.maxAge ? props.cors.maxAge : 0,
+          },
+        }
+      : {};
 
-    this.api = new ApiGatewayRestApi(this, "api", {
+    this.api = new Apigatewayv2Api(this, "api", {
       name: ResourceNames.generateName(this, NAME_OPTS),
-      // Lazy generation of the api spec because routes can be added after the API is created
+      protocolType: "HTTP",
       body: Lazy.stringValue({
         produce: () => {
-          const injectGreedy404Handler = (openApiSpec: OpenApiSpec) => {
-            openApiSpec.paths = {
-              ...openApiSpec.paths,
-              ...defaultResponse,
-            };
-            return openApiSpec;
-          };
-          return JSON.stringify(injectGreedy404Handler(props.apiSpec));
+          props.apiSpec = "random assignment so compiler doesn't complain";
```

Comment:
?

Reply:
Didn't want to comment that, and compiler throws errors if you have unused stuff in the code.

Reply:
Some leftover code from previous tests.